main.tf

```diff
@@ -262,7 +262,7 @@ resource "aws_lambda_event_source_mapping" "this" {
 
     content {
       filter {
-        pattern = lookup(filter_criteria, "pattern", null)
+        pattern = lookup(each.value["filter_criteria"], "pattern", null)
```

Comment:
I think instead we should change line 261 to

```hcl
for_each = try(each.value.filter_criteria, [])
```

which should make 265 work as intended without changes (I think, haven't tried myself yet)

Reply:
@bryantbiggs This module is still full of `lookup()` expressions, so using `try()` is not so critical here.

Reply:
I wanted to follow what's done elsewhere in this module. I can change it if you want though.